src/router.ts

```diff
@@ -325,6 +327,26 @@ export class OpenAPIRouter<D extends Document = Document> {
               const commaParsed = parseQuery(commaQueryString, { comma: true });
               query[queryParam] = commaParsed[queryParam];
             }
+            const schemaType = (parameter.schema as PickVersionElement<D, OpenAPIV3.SchemaObject, OpenAPIV3_1.SchemaObject>)?.type;
+            if (Array.isArray(query[queryParam])) {
+              query[queryParam] = (query[queryParam] as any[]).map((par) => {
```

Comment:
Not a big fan of the `any[]` here.

Maybe let's do something like:

```
type ParamValue = string | number | boolean

export interface ParsedRequest extends Request {
  params: {
    [key: string]: ParamValue | ParamValue[];
  };
  cookies: {
    [key: string]: ParamValue | ParamValue[];
  };
  query: {
    [key: string]: ParamValue | ParamValue[];
  };
  requestBody: any;
}
```

Then I believe we don't need to cast the type here at all. `Array.isArray` should be a sufficent Type Guard